src/containers/LessonsContainer/LessonsContainer.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Box, Grid, TextField, Typography } from '@material-ui/core';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import DateFnsUtils from '@date-io/date-fns';
+import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { makeStyles } from '@material-ui/core/styles';
+import { fetchGroups } from '../../store/actions/groupsAction';
+import { fetchSubjects } from '../../store/actions/subjectsAction';
+import { getTeachersBySubject } from '../../store/actions/teachersActions';
+import { getWeekdates } from '../../helpers/helpers';
+
+const useStyles = makeStyles(() => ({
+  lessonsData: {
+    margin: 50,
+    paddingLeft: 100,
+    paddingRight: 400,
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  autocomplete: {
+    marginBottom: 25,
+  },
+  dateBox: {
+    textAlign: 'center',
+  },
+  dateText: {
+    marginBottom: 5,
+    textAlign: 'start',
+    marginLeft: 20,
+  },
+  autoComplTeacher: {
+    marginTop: 20,
+  },
+}));
+
+const LessonsContainer = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchGroups());
+    dispatch(fetchSubjects());
+  }, [dispatch]);
+
+  const { groups } = useSelector((state) => state.groups);
+  const { subjects } = useSelector((state) => state.subjects);
+  const { teachersBySubject } = useSelector((state) => state.teachers);
+
+  const [group, setGroup] = useState(null);
+  const [subject, setSubject] = useState(null);
+  const [teacher, setTeacher] = useState(null);
+  const [selectedDate, setSelectedDate] = useState(new Date());
+
+  const copySelectedDate = new Date(selectedDate);
+  getWeekdates(copySelectedDate);
+
+  const isInitialMount = useRef(true);
+  useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
```

Comment:
1) для чего используется это условие? isInitialMount всегда false
2) чем отличается isInitialMount от без него
```
  useEffect(() => {
    dispatch(getTeachersBySubject(lesson.subjectId));
  }, [lesson]);
```
3) надо поставить условие  lesson.subjectId &&  впереди dispatch, если внутри, то когда нажимаешь на X в выборе Предмета,  то появляется ошибка 
![image](https://user-images.githubusercontent.com/45433350/127911634-c6a14f2e-06be-41b3-b323-e44ce153d410.png)

```suggestion
  useEffect(() => {
    lesson.subjectId && dispatch(getTeachersBySubject(lesson.subjectId));
  }, [lesson]);
```

Reply:
initialMount и это условие выполняется чтобы, этот юзефект не вызывался при рендере, а только вызовался только при изменении lesson. Убрал усовие и написал subjectId && перед dispatch.